code/modules/atmospherics/machinery/bluespace_vendor.dm

```diff
@@ -0,0 +1,278 @@
+/obj/item/wallframe/bluespace_vendor_mount
+	name = "bluespace vendor wall mount"
+	desc = "Used for placing bluespace vendors."
+	icon = 'icons/obj/atmospherics/components/bluespace_gas_selling.dmi'
+	icon_state = "bluespace_vendor_open"
+	result_path = /obj/machinery/bluespace_vendor/built
+
+///Defines for the mode of the vendor
+#define BS_MODE_OFF 1
+#define BS_MODE_IDLE 2
+#define BS_MODE_PUMPING 3
+#define BS_MODE_OPEN 4
+
+/obj/machinery/bluespace_vendor
+	icon = 'icons/obj/atmospherics/components/bluespace_gas_selling.dmi'
+	icon_state = "bluespace_vendor_off"
+	name = "Bluespace Gas Vendor"
+	desc = "Sells gas tanks with custom mixes for all the family!"
+
+	max_integrity = 300
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 100, BOMB = 0, BIO = 100, RAD = 100, FIRE = 80, ACID = 30)
+	layer = OBJ_LAYER
+
+	///The bluespace sender that this vendor is connected to
+	var/obj/machinery/atmospherics/components/unary/bluespace_sender/connected_machine
+	///Amount of usable tanks inside the machine
+	var/empty_tanks = 10
+	///Reference to the current in use tank to be filled
+	var/obj/item/tank/internal_tank
+	///Path of the gas selected from the UI to be pumped inside the tanks
+	var/selected_gas
+	///Is the vendor trying to move gases from the network to the tanks?
+	var/pumping = FALSE
+	///Has the user prepared a tank to be filled with gases?
+	var/inserted_tank = FALSE
+	///Amount of the tank already filled with gas (from 0 to 100)
+	var/tank_filling_amount = 0
+	///Base price of the tank
+	var/tank_cost = 10
+	///Stores the current price of the gases inside the tank
+	var/gas_price = 0
+	///Helper for mappers, will automatically connect to the sender (ensure to only place one sender per map)
+	var/map_spawned = TRUE
+	///Base icon name for updating the appearance
+	var/base_icon = "bluespace_vendor"
+	///Current operating mode of the vendor
+	var/mode = BS_MODE_OFF
+
+//The one that the players make
+/obj/machinery/bluespace_vendor/built
+	map_spawned = FALSE
+	mode = BS_MODE_OPEN
+
+/obj/machinery/bluespace_vendor/north //Pixel offsets get overwritten on New()
+	dir = SOUTH
+	pixel_y = 30
+
+/obj/machinery/bluespace_vendor/south
+	dir = NORTH
+	pixel_y = -30
+
+/obj/machinery/bluespace_vendor/east
+	dir = WEST
+	pixel_x = 30
+
+/obj/machinery/bluespace_vendor/west
+	dir = EAST
+	pixel_x = -30
+
+/obj/machinery/bluespace_vendor/New(loc, ndir, nbuild)
+	. = ..()
+	if(ndir)
+		setDir(ndir)
+
+	if(nbuild)
+		panel_open = TRUE
+		pixel_x = (dir & 3)? 0 : (dir == 4 ? -30 : 30)
+		pixel_y = (dir & 3)? (dir == 1 ? -30 : 30) : 0
+
+	update_appearance()
+
+/obj/machinery/bluespace_vendor/Initialize()
+	. = ..()
+	AddComponent(/datum/component/payment, tank_cost, SSeconomy.get_dep_account(ACCOUNT_ENG), PAYMENT_ANGRY)
+
+/obj/machinery/bluespace_vendor/LateInitialize()
+	. = ..()
+	if(!map_spawned)
+		return
+	for(var/obj/machinery/atmospherics/components/unary/bluespace_sender/sender in GLOB.machines)
+		if(!sender)
```

Comment:
It wont be null

Reply:
i'll remove it, but can't it be null if the sender doesn't get placed on the map?

Reply:
If there are none of that machine when you do a filtered loop like this, then the code inside that loop will run 0 times.